### PR TITLE
CORTX-29735: Collect rgw core files as a part of rgw support bundle script

### PR DESCRIPTION
**Pull request by rohit-k-dwivedi**
Signed-off-by: Rohit Dwivedi <rohit.k.dwivedi@seagate.com>
JIRA : https://jts.seagate.com/browse/CORTX-29735
# Problem Statement
- core dumps are not captured in rgw support bundle

# Design
-  CRASHDUMP path is updated
# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/rgw/support/rgw_support_bundle [hunk 1]
```diff
@@ -78,7 +77,8 @@ class RGWSupportBundle:
                 if regex.match(file):
                     infile = os.path.join(log_dir, file)
                     outfile = os.path.join(RGWSupportBundle._tmp_src, file)
-                    shutil.copyfile(infile, outfile)
+                    if os.path.isfile(infile):
```
Comment (rohit-k-dwivedi) [inline src/rgw/support/rgw_support_bundle:80]: this avoids error if any directory matches the regex `regex = re.compile('(rgw*)|(radosgw*)')` in above line number 75

File: src/rgw/support/rgw_support_bundle [hunk 2]
```diff
@@ -91,8 +91,9 @@ class RGWSupportBundle:
 
         # copy ceph crash-dump files
         if coredumps:
-            if os.path.exists(CRASHDUMP_DIR):
-                shutil.copytree(CRASHDUMP_DIR,\
+            crash_dump_dir = os.path.join(log_dir, 'rgw_debug')
+            if os.path.exists(crash_dump_dir):
+                shutil.copytree(crash_dump_dir,\
                     os.path.join(RGWSupportBundle._tmp_src, 'crash'))
```
Comment (sachitanands) [inline src/rgw/support/rgw_support_bundle:97]: could you please update name "crash" -> "core_files"

**Review by sachitanands — APPROVED** [review]
rest of the changes looks good

File: src/rgw/support/rgw_support_bundle (hunk 1 above)
Comment (cdeshmukh) [reply src/rgw/support/rgw_support_bundle:80]: this is exactly where CORTX-32068 reported failure exists

File: src/rgw/support/rgw_support_bundle (hunk 2 above)
Comment (rohit-k-dwivedi) [reply src/rgw/support/rgw_support_bundle]: done

Comment (cdeshmukh) [reply src/rgw/support/rgw_support_bundle]: why name being changed from rgw_debug to crash or core_files ? we need to retain same

Comment (sachitanands) [reply src/rgw/support/rgw_support_bundle]: okay. I though core will be more easier which is inline with motr cores as well.
but should be okay to use 'rgw_debug' as well.

Comment (sachitanands) [reply src/rgw/support/rgw_support_bundle]: @rohit-k-dwivedi  pls revert this change..lets use same name as 'rgw_debug'

**Comment by rahul-tripathi-git** [comment]
@rohit-k-dwivedi 
Please try out deployment as well.

File: src/rgw/support/rgw_support_bundle (hunk 2 above)
Comment (rohit-k-dwivedi) [reply src/rgw/support/rgw_support_bundle]: done @cdeshmukh @sachitanands

**Review by cdeshmukh — APPROVED** [review]

**Comment by rohit-k-dwivedi** [comment]
tested deployment with image: 
```
cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6532-custom-ci 
cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-6532-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6532-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6532-custom-ci
```

created a dummy file `core1` under `/etc/cortx/log/rgw/(machine-id)/rgw_debug`

```
/root/deploy-scripts/k8_cortx_cloud/logs-cortx-cloud-2022-06-07_01-52/bundle-logs-cortx-server-ssc-vm-rhev4-2782-6bd8c64454-7xpsv-2022-06-07_01-52/bundle-logs-cortx-server-ssc-vm-rhev4-2782-6bd8c64454-7xpsv-2022-06-07_01-52/rgw
[root@ssc-vm-rhev4-2782 rgw]# ll
total 5132
drwxr-xr-x 2 root root    4096 Jun  7 01:47 addb_files-0x7200000000000001:0x30
-rw-r--r-- 1 root root    1764 Jun  7 01:55 cortx_rgw.conf
-rw-r--r-- 1 root root   16884 Jun  7 01:55 cortx-rpms
drwxr-xr-x 2 root root    4096 Jun  7 01:55 motr_trace_files
-rw-r--r-- 1 root root    6836 Jun  7 01:55 radosgw-admin.log
-rw-r--r-- 1 root root  308912 Jun  7 01:55 rgw-1.log
-rw-r--r-- 1 root root 4880411 Jun  7 01:55 rgw_bundle-logs-cortx-server-ssc-vm-rhev4-2782-6bd8c64454-7xpsv-2022-06-07_01-52_5e98334db1ee74ec59641ec8ceb75605_rgw.tar.gz
drwxr-xr-x 2 root root    4096 Jun  7 01:51 rgw_debug
-rw-r--r-- 1 root root    9125 Jun  7 01:55 rgw_setup.log
-rw-r--r-- 1 root root       0 Jun  7 01:55 rgw_startup.log
-rw-r--r-- 1 root root     102 Jun  7 01:55 rgw_support_bundle.log
[root@ssc-vm-rhev4-2782 rgw]# ll rgw_debug/
total 0
-rw-r--r-- 1 root root 0 Jun  7 01:47 1
-rw-r--r-- 1 root root 0 Jun  7 01:51 core1

```

**Pull request merged**